nselib/http.lua

```diff
@@ -1571,7 +1571,9 @@ function parse_redirect(host, port, path, response)
   if ( not(u.host) ) then
     -- we're dealing with a relative url
     u.host = stdnse.get_hostname(host)
-    u.path = ((u.path:sub(1,1) == "/" and "" ) or "/" ) .. u.path -- ensuring leading slash
+    if ( u.path:sub(1,1) ~= "/") then
```

Comment:
I do not believe that you need the `if` here at all because `url.absolute()` already does the right thing if the second parameter is an absolute path.

Secondly, you could also merge in the following `if` block (found a few lines later in the code):
```lua
if ( not(u.path) ) then
  u.path = "/"
end
```
...because the block is a bit too late in the code. The check for absence of path should be performed earlier in the routine. At the end you might end up with something as simple as:
```lua
u.path = url.absolute(path, u.path or "/")
```